### DISCUSSION TITLE
Maven deploy fails with status 422 unprocessable entity

### DISCUSSION THREAD

**Discussion by mikenakis** [community/community#23474] [Packages] [answered]
<p>I have this repository: <a href="https://github.com/mikenakis/Bathyscaphe" class="inline-onebox" rel="noopener nofollow ugc">GitHub - mikenakis/Bathyscaphe: Deep immutability and thread-safety assessment for Java objects</a> in which both my CI-Workflow and my Release-Workflow work fine.</p>
<p>Now I am trying to extract a sub-module from that repository into a new repository.</p>
<p>So, I created this repository: <a href="https://github.com/mikenakis/BathyscapheClaims" class="inline-onebox" rel="noopener nofollow ugc">GitHub - mikenakis/BathyscapheClaims</a></p>
<p>The CI-Workflow works, but when I try to launch the Relase-Workflow, I get the following error:</p>
<p>Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:3.0.0-M2:deploy (default-deploy) on project bathyscaphe-claims: ArtifactDeployerException: Failed to deploy artifacts: Could not transfer artifact io.github.mikenakis:bathyscaphe-claims:jar:1.3 from/to github (<a href="https://maven.pkg.github.com/mikenakis/BathyscapheClaims" rel="noopener nofollow ugc">https://maven.pkg.github.com/mikenakis/BathyscapheClaims</a>): transfer failed for <a href="https://maven.pkg.github.com/mikenakis/BathyscapheClaims/io/github/mikenakis/bathyscaphe-claims/1.3/bathyscaphe-claims-1.3.jar" rel="noopener nofollow ugc">https://maven.pkg.github.com/mikenakis/BathyscapheClaims/io/github/mikenakis/bathyscaphe-claims/1.3/bathyscaphe-claims-1.3.jar</a>, status: 422 Unprocessable Entity → [Help 1]</p>
<p>I have already seen the issues that appear to be similar; most of them have no answer, one of them (<a href="https://github.community/t/maven-deploy-fails-with-http-422-unprocessable-entity/119670" class="inline-onebox">Maven deploy fails with HTTP 422 Unprocessable Entity</a>) has an extensive discussion, but it does not seem to apply in my case.</p>
<p>Can someone please help me?  It is probably something silly, but I only have one pair of eyes, while this obviously needs two!  (-:=</p>
<p>Cheers!</p>

**Comment by mikenakis** [comment, marked as answer]
<p>I figured this one out too by myself.<br>
(Hey community, you are not helping much!)</p>
<p>What happened is that GitHub apparently has some monstrous bug which causes artifact deployment to fail if some other repository (only by the same owner, I hope!) has already deployed an artifact with the exact same name.</p>
<p>In this case, one repository was called Bathyscaphe, the other was called BathyscapheClaims, but each repository was trying to deploy an artifact called bathyscaphe-claims.</p>
<p>I changed the name of the artifact in the second repository from bathyscaphe-claims to bathyscaphe-claims2 and it worked.</p>
<p>The situation was certainly made worse by this contemptible monstrosity that goes by the name of “maven”, which never gives any meaningful error message whatsoever when the slightest thing goes wrong.  I mean, “unprocessable entity”? Really? It must have been some very special kind of idiot who came up with this error message.</p>

**Reply by dvag-martin-anlauf** [reply]
Thanks for the analysis. I encountered the same problem.

**Reply by queenofcorgis** [reply]
Way to self-solve @mikenakis ⭐, glad it was able to help you, @dvag-martin-anlauf ❤️

**Reply by casanella** [reply]
@mikenakis u make my day more than 3hours trying to figure out why could not deploy, after changing name it works

Lot of kudos!!!

**Reply by mhail** [reply]
Hit the same brick wall. Your ticket helped me. Deleting the other package that was using the same artifactId and re-triggering a build was successful.

**Reply by serializer** [reply]
is there no way to unlink it so that I don't have to change the artifact name?

**Reply by IsmailMarmoush** [reply]
@tronku yes, I just removed all "packages" at the old repo, and when started uploading again it worked.

**Reply by ebonnet31** [reply]
yep, same problem. Another repo with same package target name. I would have searched for hours without this post !

**Reply by eferraris** [reply]
I had the exact same problem! I removed the package from the other repo and it worked just fine! Thanks for sharing the solution!

**Reply by emitchel** [reply]
Changing the groupId also bypasses the issue, which was an easier option for our team than deleting all packages in the other repo.

**Reply by janiis-scott** [reply]
This post is still saving the day! Just ran into this exact issue.

**Comment by anhtester** [comment]
I faced the same problem and haven't found a solution yet.
Someone has solved it please help me.

![image](https://user-images.githubusercontent.com/87883620/211014224-82e60c66-d38e-4637-92e0-6b45ffaaf5a4.png)

**Reply by anhtester** [reply]
Yeah i fixed it when changed artifactId to lowercase.

**Reply by Acesif** [reply]
Thanks. That worked, such a strange bug.

**Comment by dmitriigriazin** [comment]
Confirm, in our case it was just the file path validation error: it should be lowercase to be able to push to github packages.

**Reply by anhtester** [reply]
Yep !!

**Reply by RSHKwee** [reply]
yep! Why is the error message so crapy……😉

**Reply by ColorAutumn** [reply]
I tried all afternoon and couldn't think of a reason like this!

**Reply by harry-cong-wadiz** [reply]
Wow... just trying to move to github packages from jetbrains space.  Why do I change artifactId to lower case? 
Anyway, I also succeeded deployment changing artifactId to lower case.
Many thanks!

**Comment by javedsshaikh** [comment]
@mikenakis, thank you very much. You saved the day and frustration. Shows the importance of proper error messages, which unfortunately is ignored even by such widely used tool.

**Comment by KPragadeesh** [comment]
@mikenakis Thanks for this thread. Confirmed. Yes it should be in lowercase. It worked for me.
